src/cargo/core/workspace.rs

```diff
@@ -316,9 +318,16 @@ impl<'cfg> Workspace<'cfg> {
         };
 
         if let Some(list) = members {
+            let root = root_manifest.parent().unwrap();
+
+            let mut expanded_list = Vec::new();
             for path in list {
-                let root = root_manifest.parent().unwrap();
-                let manifest_path = root.join(path).join("Cargo.toml");
+                let expanded_paths = expand_member_path(&path, root)?;
+                expanded_list.extend(expanded_paths);
+            }
+
+            for path in expanded_list {
+                let manifest_path = path.join("Cargo.toml");
```

Comment:
Will this error if there's a path **without** `Cargo.toml` matched by the glob? I would say that it's OK to bail in this case, but that just skipping that path would be probably more useful: I often have various assorted folders alongside the workspace packages. 

In any case, I think a test case for this would be useful :)